secretsclient/client.go

```diff
@@ -143,16 +185,18 @@ type client struct {
 	sshKey                string
 	lastUsedPublicKeyPath string
 	sshAgent              agent.ExtendedAgent
+	warnFunc              func(string, ...interface{})
```

Comment:
Maybe it's simpler to just pass the entire `ConsoleLogger` object? But not a strong opinion.

Reply:
I went with this as it gives us an option to move this client into an outside library down the road.

Reply:
Good idea